File: backend/SmartTradePlatformBackend/src/main/java/smarttrade/backend/service/itemService.java
package smarttrade.backend.service;


import smarttrade.backend.entities.itemEntity;
import smarttrade.backend.repository.itemRepo;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Service
public class itemService {
    itemRepo itemRepo;
    itemService(itemRepo itemRepo){
        this.itemRepo=itemRepo;
    }
    public List<itemEntity> GetAllItems() {
        
        return itemRepo.findAll();
    }

    public Optional<itemEntity> GetItem(Long itemId) {
        return itemRepo.findById(itemId);
    }

    public itemEntity addItems(itemEntity item) {
        return itemRepo.save(item);
    }


    public itemEntity updateItem(Long itemId, itemEntity item) {
        item.setItemId(itemId);
        return itemRepo.save(item);
    }

    public void deleteItem(Long itemId) {
        itemRepo.deleteById(itemId);
    }

    public List<itemEntity> getNearbyItems(double lat, double lng, double radiusKm) {
        double radiusMeters = radiusKm * 1000;
        return itemRepo.findNearbyItems(lat, lng, radiusMeters);
    }

    public List<itemEntity> searchItems(String category, List<String> labels, String name) {
        return itemRepo.searchItems(category, name);
    }
    public void updateAvailability(Long itemId, boolean available) {
        Optional<itemEntity> optionalItem = itemRepo.findById(itemId);
        if (optionalItem.isEmpty()) {
            throw new IllegalArgumentException("Item not found");
        }

        itemEntity item = optionalItem.get();
        item.setAvailable(available);
        itemRepo.save(item);
    }

}

File: frontend/SmartTradePlatformFrontEnd/src/services/itemService.js
import api from './api';

export const itemService = {
  // Get all items
  getAllItems: async () => {
    const response = await api.get('/items');
    return response.data;
  },

  // Get item by ID
  getItem: async (itemId) => {
    const response = await api.get(`/items/${itemId}`);
    return response.data;
  },

  // Create item
  createItem: async (itemData) => {
    const response = await api.post('/items', itemData);
    return response.data;
  },

  // Update item
  updateItem: async (itemId, itemData) => {
    const response = await api.patch(`/items/${itemId}`, itemData);
    return response.data;
  },

  // Delete item
  deleteItem: async (itemId) => {
    await api.delete(`/items/${itemId}`);
  },

  // Get nearby items
  getNearbyItems: async (lat, lng, radiusKm = 10) => {
    const response = await api.get('/items/nearby', {
      params: { lat, lng, radiusKm },
    });
    return response.data;
  },

  // Search items
  searchItems: async (filters = {}) => {
    const params = {};
    if (filters.category) params.category = filters.category;
    if (filters.name) params.name = filters.name;
    if (filters.labels && filters.labels.length > 0) {
      params.labels = filters.labels;
    }
    const response = await api.get('/items/search', { params });
    return response.data;
  },

  // Toggle item availability
  toggleAvailability: async (itemId, available) => {
    await api.patch(`/items/${itemId}/availability`, null, {
      params: { available },
    });
  },
};

